corehq/blobs/pillow.py

```diff
@@ -19,46 +14,28 @@
 
 class BlobDeletionProcessor(PillowProcessor):
 
-    def __init__(self, blob_db, bucket_base):
+    def __init__(self, blob_db):
         super(BlobDeletionProcessor, self).__init__()
         self.blob_db = blob_db
-        self.bucket_base = bucket_base
 
     def process_change(self, pillow_instance, change):
         if change.deleted:
-            bucket = join(self.bucket_base, change.id)
+            bucket = "{}/{}".format(change.meta.data_source_name, change.id)
```

Comment:
sorry I think i am creating confusion here and possibly github is not the best place for this conversation. here's my quick summary of the situation.

I think the ideal future state is:

- pillows publish both soft and hard deletions and differentiate between them
- blobs are deleted on all hard deletions
- blobs are not deleted on all soft deletions (including forms)

I think the current state is:

- pillows inconsistently publish hard deletions (couch pillows do, kafka pillows mostly don't)
- hard and soft deletions are not differentiable
- blobs are deleted on hard deletions in couch pillows

I think this PR changes the last thing to:

- blobs are deleted on hard and soft deletions that happen to get published to the `META` and `APP` topics (which may not be all of them because of the first bullet in "current state".

which is probably a net worse change. So I'm tempted to propose as next steps:

- we put in the backlog to fix the hard/soft deletion publishing
- when that's done we come back to this and it should be ready to drop in.

thoughts?